### ISSUE TITLE
Error unbinding existing q handler from AF_INET protocol family: invalid argument

### ISSUE THREAD

**Issue by legraphista** [evilsocket/opensnitch#912]
**Describe the bug**
Installed the daemon, followed the [Install](https://github.com/evilsocket/opensnitch/wiki/Installation) and [How it works](https://github.com/evilsocket/opensnitch/wiki/Dependencies-and-how-it-works) instructions, but when running `sudo /usr/bin/opensnitchd -rules-path /etc/opensnitchd/rules -debug` i get the following log:
```
[2023-04-18 08:59:33]  IMP  Starting opensnitch-daemon v1.5.2
[2023-04-18 08:59:33]  INF  Loading rules from /etc/opensnitchd/rules ...
[2023-04-18 08:59:33]  DBG  Starting 16 workers ...
[2023-04-18 08:59:33]  DBG  Worker #4 started.
[2023-04-18 08:59:33]  DBG  Worker #12 started.
[2023-04-18 08:59:33]  DBG  Worker #15 started.
[2023-04-18 08:59:33]  DBG  Stats worker #3 started.
[2023-04-18 08:59:33]  DBG  Worker #7 started.
[2023-04-18 08:59:33]  DBG  Rules watcher started on path /etc/opensnitchd/rules ...
[2023-04-18 08:59:33]  DBG  Stats worker #0 started.
[2023-04-18 08:59:33]  DBG  Stats worker #1 started.
[2023-04-18 08:59:33]  DBG  Worker #8 started.
[2023-04-18 08:59:33]  DBG  Worker #5 started.
[2023-04-18 08:59:33]  DBG  Worker #13 started.
[2023-04-18 08:59:33]  DBG  Worker #6 started.
[2023-04-18 08:59:33]  DBG  Worker #0 started.
[2023-04-18 08:59:33]  DBG  Stats worker #2 started.
[2023-04-18 08:59:33]  DBG  Worker #14 started.
[2023-04-18 08:59:33]  DBG  Worker #2 started.
[2023-04-18 08:59:33]  DBG  Worker #1 started.
[2023-04-18 08:59:33]  DBG  Worker #10 started.
[2023-04-18 08:59:33]  DBG  Worker #3 started.
[2023-04-18 08:59:33]  DBG  Worker #11 started.
[2023-04-18 08:59:33]  DBG  Worker #9 started.
[2023-04-18 08:59:33]  WAR  Is opensnitchd already running?
[2023-04-18 08:59:33]  !!!  Error while creating queue #0: Error unbinding existing q handler from AF_INET protocol family: invalid argument
```
Include the following information:
 - OpenSnitch version: 1.5.2
 - OS: PopOS
 - Version: Pop!_OS 22.04 LTS
 - Window Manager: GNOME
 - Kernel version: `Linux stefan-mobile-desktop 6.2.6-76060206-generic #202303130630~1680814622~22.04~3850312 SMP PREEMPT_DYNAMIC Thu A x86_64 x86_64 x86_64 GNU/Linux`

**Additional context**

`sudo ps -aux | grep [s]nitch`
```
stefan    202834  0.0  0.0 308744 21760 pts/0    SN+  10:49   0:00 journalctl -f -u opensnitch.service
stefan    211450  0.0  0.0  22848  4352 pts/9    SN+  11:10   0:00 systemctl show opensnitch
```

`sudo grep NFT /boot/config-$(uname -r)`

<details>
<summary>Details</summary>

```
CONFIG_NFT_NUMGEN=m
CONFIG_NFT_CT=m
CONFIG_NFT_FLOW_OFFLOAD=m
CONFIG_NFT_CONNLIMIT=m
CONFIG_NFT_LOG=m
CONFIG_NFT_LIMIT=m
CONFIG_NFT_MASQ=m
CONFIG_NFT_REDIR=m
CONFIG_NFT_NAT=m
CONFIG_NFT_TUNNEL=m
CONFIG_NFT_QUEUE=m
CONFIG_NFT_QUOTA=m
CONFIG_NFT_REJECT=m
CONFIG_NFT_REJECT_INET=m
CONFIG_NFT_COMPAT=m
CONFIG_NFT_HASH=m
CONFIG_NFT_FIB=m
CONFIG_NFT_FIB_INET=m
CONFIG_NFT_XFRM=m
CONFIG_NFT_SOCKET=m
CONFIG_NFT_OSF=m
CONFIG_NFT_TPROXY=m
CONFIG_NFT_SYNPROXY=m
CONFIG_NFT_DUP_NETDEV=m
CONFIG_NFT_FWD_NETDEV=m
CONFIG_NFT_FIB_NETDEV=m
CONFIG_NFT_REJECT_NETDEV=m
CONFIG_NFT_REJECT_IPV4=m
CONFIG_NFT_DUP_IPV4=m
CONFIG_NFT_FIB_IPV4=m
CONFIG_NFT_REJECT_IPV6=m
CONFIG_NFT_DUP_IPV6=m
CONFIG_NFT_FIB_IPV6=m
CONFIG_NFT_BRIDGE_META=m
CONFIG_NFT_BRIDGE_REJECT=m
CONFIG_NFTL=m
CONFIG_NFTL_RW=y
CONFIG_INFTL=m
```

</details>

`sudo lsmod | grep -i queue`

```
<empty>
```

`sudo cat /proc/net/ip_tables_targets`
```
DNAT
SNAT
DNAT
SNAT
DNAT
SNAT
MASQUERADE
ERROR
```
`sudo cat /proc/net/ip_tables_targets`
```
DNAT
SNAT
DNAT
SNAT
DNAT
SNAT
MASQUERADE
ERROR
```
`sudo cat /proc/net/netfilter/nfnetlink_queue`
```
cat: /proc/net/netfilter/nfnetlink_queue: No such file or directory
```
`sudo grep NFQUEUE /boot/config-$(uname -r)`
```
CONFIG_NETFILTER_XT_TARGET_NFQUEUE=m
```
`dpkg -l 'libnetfilter-queue*'`
```
Desired=Unknown/Install/Remove/Purge/Hold
| Status=Not/Inst/Conf-files/Unpacked/halF-conf/Half-inst/trig-aWait/Trig-pend
|/ Err?=(none)/Reinst-required (Status,Err: uppercase=bad)
||/ Name                      Version      Architecture Description
+++-=========================-============-============-=================================
ii  libnetfilter-queue1:amd64 1.0.5-2      amd64        Netfilter netlink-queue library
```
`sudo grep CONFIG_NETFILTER_NETLINK_QUEUE /boot/config-$(uname -r)`
```
CONFIG_NETFILTER_NETLINK_QUEUE=m
```
`sudo grep CONFIG_NETFILTER_XT_TARGET_NFQUEUE /boot/config-$(uname -r)`
```
CONFIG_NETFILTER_XT_TARGET_NFQUEUE=m
```
`sudo grep CONFIG_KPROBES /boot/config-$(uname -r)`
```
CONFIG_KPROBES=y
CONFIG_KPROBES_ON_FTRACE=y
```
`sudo nft list ruleset`

<details>
<summary>Details</summary>

```
table ip nat {
        chain POSTROUTING {
                type nat hook postrouting priority srcnat; policy accept;
                oifname != "docker0" ip saddr 172.17.0.0/16 counter packets 0 bytes 0 masquerade 
                oifname != "br-9611a650fee0" ip saddr 172.18.0.0/16 counter packets 8 bytes 480 masquerade 
        }

        chain PREROUTING {
                type nat hook prerouting priority dstnat; policy accept;
                fib daddr type local counter packets 1 bytes 328 jump DOCKER
        }

        chain OUTPUT {
                type nat hook output priority -100; policy accept;
                ip daddr != 127.0.0.0/8 fib daddr type local counter packets 0 bytes 0 jump DOCKER
        }

        chain DOCKER {
                iifname "docker0" counter packets 0 bytes 0 return
                iifname "br-9611a650fee0" counter packets 0 bytes 0 return
        }
}
table ip filter {
        chain DOCKER {
        }

        chain DOCKER-ISOLATION-STAGE-1 {
                iifname "docker0" oifname != "docker0" counter packets 0 bytes 0 jump DOCKER-ISOLATION-STAGE-2
                iifname "br-9611a650fee0" oifname != "br-9611a650fee0" counter packets 0 bytes 0 jump DOCKER-ISOLATION-STAGE-2
                counter packets 0 bytes 0 return
        }

        chain FORWARD {
                type filter hook forward priority filter; policy drop;
                counter packets 0 bytes 0 jump DOCKER-USER
                counter packets 0 bytes 0 jump DOCKER-ISOLATION-STAGE-1
                oifname "docker0" ct state related,established counter packets 0 bytes 0 accept
                oifname "docker0" counter packets 0 bytes 0 jump DOCKER
                iifname "docker0" oifname != "docker0" counter packets 0 bytes 0 accept
                iifname "docker0" oifname "docker0" counter packets 0 bytes 0 accept
                oifname "br-9611a650fee0" ct state related,established counter packets 93 bytes 55370 accept
                oifname "br-9611a650fee0" counter packets 0 bytes 0 jump DOCKER
                iifname "br-9611a650fee0" oifname != "br-9611a650fee0" counter packets 102 bytes 16487 accept
                iifname "br-9611a650fee0" oifname "br-9611a650fee0" counter packets 0 bytes 0 accept
        }

        chain DOCKER-USER {
                counter packets 195 bytes 71857 return
        }

        chain DOCKER-ISOLATION-STAGE-2 {
                oifname "docker0" counter packets 0 bytes 0 drop
                oifname "br-9611a650fee0" counter packets 0 bytes 0 drop
                counter packets 0 bytes 0 return
        }
}

```

</details>

`sudo iptables -V`
```
iptables v1.8.7 (nf_tables)
```

`sudo iptables -L INPUT`
```
Chain INPUT (policy ACCEPT)
target     prot opt source               destination        
```

`sudo iptables -t mangle -L OUTPUT`
```
Chain OUTPUT (policy ACCEPT)
target     prot opt source               destination       
```

**Comment by gustavo-iniguez-goya** [comment]
Hi @legraphista ,

Thank you very much for detailed report :heart: 

> cat: /proc/net/netfilter/nfnetlink_queue: No such file or directory

See if you have the nfqueue netfilter modules loaded: `lsmod|grep -i queue`

The modules `xt_NFQUEUE` and `nfnetlink_queue` should be loaded. If they aren't, load them manually and start the daemon again.

Also check if the module `ip_queue` is loaded, I doubt it, but if it is, unload it. On Ubuntu Ports it prevented the daemon from working.

**Comment by gustavo-iniguez-goya** [comment]
I've reviewed my notes on these errors, and this one in particular is usually caused when `nfnetlink_queue` is not loaded.
So maybe there's another condition causing this error.

Could you launch the daemon with another queue number?:
`$ sudo /usr/bin/opensnitchd -rules-path /etc/opensnitchd/rules/ -queue-num 2 -debug`

**Comment by legraphista** [comment]
Looks like on `-queue-num 2` it works

<details>
<summary>Log</summary>

```
$ sudo /usr/bin/opensnitchd -rules-path /etc/opensnitchd/rules/ -queue-num 2 -debug
[2023-04-20 07:17:41]  IMP  Starting opensnitch-daemon v1.5.2
[2023-04-20 07:17:41]  INF  Loading rules from /etc/opensnitchd/rules ...
[2023-04-20 07:17:41]  DBG  Starting 16 workers ...
[2023-04-20 07:17:41]  DBG  Stats worker #1 started.
[2023-04-20 07:17:41]  DBG  Rules watcher started on path /etc/opensnitchd/rules ...
[2023-04-20 07:17:41]  DBG  Worker #6 started.
[2023-04-20 07:17:41]  DBG  Stats worker #0 started.
[2023-04-20 07:17:41]  DBG  Worker #4 started.
OK: libnetfiler_queue supports nfq_get_uid
[2023-04-20 07:17:41]  DBG  Worker #9 started.
[2023-04-20 07:17:41]  DBG  Stats worker #2 started.
OK: libnetfiler_queue supports nfq_get_uid
[2023-04-20 07:17:41]  DBG  Worker #10 started.
[2023-04-20 07:17:41]  DBG  Worker #2 started.
[2023-04-20 07:17:41]  DBG  Worker #7 started.
[2023-04-20 07:17:41]  DBG  Worker #8 started.
[2023-04-20 07:17:41]  DBG  Worker #5 started.
[2023-04-20 07:17:41]  DBG  Worker #1 started.
[2023-04-20 07:17:41]  DBG  Worker #0 started.
[2023-04-20 07:17:41]  DBG  Worker #11 started.
[2023-04-20 07:17:41]  DBG  Worker #12 started.
[2023-04-20 07:17:41]  DBG  Worker #14 started.
[2023-04-20 07:17:41]  DBG  Worker #15 started.
[2023-04-20 07:17:41]  DBG  Worker #13 started.
[2023-04-20 07:17:41]  DBG  Stats worker #3 started.
[2023-04-20 07:17:41]  DBG  Worker #3 started.
[2023-04-20 07:17:42]  INF  Running on netfilter queue #2 ...
[2023-04-20 07:17:42]  DBG  new connection tcp => 22:192.168.0.151 -> 192.168.0.205:58236 uid: 0
[2023-04-20 07:17:42]  DBG  [0/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58236 inode: 66369 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  [1/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58226 inode: 66303 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:192.168.0.151 -> 192.168.0.205:58226
[2023-04-20 07:17:42]  DBG  [2/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:0.0.0.0 -> 0.0.0.0:0 inode: 30940 - loopback: false multicast: false unspecified: true linklocalunicast: false ifaceLocalMulticast: false GlobalUni: false
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:0.0.0.0 -> 0.0.0.0:0
[2023-04-20 07:17:42]  DBG  new pid lookup took (4893): 1.244939ms
[2023-04-20 07:17:42]  DBG  [0] PID found 4893 [66369]
[2023-04-20 07:17:42]  DBG  UI is not running or busy, connected: false, running: false
[2023-04-20 07:17:42]  DBG  new connection tcp => 22:192.168.0.151 -> 192.168.0.205:58236 uid: 0
[2023-04-20 07:17:42]  DBG  [0/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58236 inode: 66369 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  [1/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58226 inode: 66303 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:192.168.0.151 -> 192.168.0.205:58226
[2023-04-20 07:17:42]  DBG  [2/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:0.0.0.0 -> 0.0.0.0:0 inode: 30940 - loopback: false multicast: false unspecified: true linklocalunicast: false ifaceLocalMulticast: false GlobalUni: false
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:0.0.0.0 -> 0.0.0.0:0
[2023-04-20 07:17:42]  DBG  Inode found in cache: 4.609µs 4893 66369 66369 192.168.0.151 22 192.168.0.205 58236
[2023-04-20 07:17:42]  DBG  [0] PID found 4893 [66369]
[2023-04-20 07:17:42]  DBG  UI is not running or busy, connected: false, running: false
[2023-04-20 07:17:42]  DBG  new connection tcp => 22:192.168.0.151 -> 192.168.0.205:58236 uid: 0
[2023-04-20 07:17:42]  DBG  [0/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58236 inode: 66369 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  [1/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58226 inode: 66303 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:192.168.0.151 -> 192.168.0.205:58226
[2023-04-20 07:17:42]  DBG  [2/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:0.0.0.0 -> 0.0.0.0:0 inode: 30940 - loopback: false multicast: false unspecified: true linklocalunicast: false ifaceLocalMulticast: false GlobalUni: false
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:0.0.0.0 -> 0.0.0.0:0
[2023-04-20 07:17:42]  DBG  Inode found in cache: 3.64µs 4893 66369 66369 192.168.0.151 22 192.168.0.205 58236
[2023-04-20 07:17:42]  DBG  [0] PID found 4893 [66369]
[2023-04-20 07:17:42]  DBG  UI is not running or busy, connected: false, running: false
[2023-04-20 07:17:42]  DBG  new connection tcp => 22:192.168.0.151 -> 192.168.0.205:58236 uid: 0
[2023-04-20 07:17:42]  DBG  [0/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58236 inode: 66369 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  [1/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58226 inode: 66303 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:192.168.0.151 -> 192.168.0.205:58226
[2023-04-20 07:17:42]  DBG  [2/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:0.0.0.0 -> 0.0.0.0:0 inode: 30940 - loopback: false multicast: false unspecified: true linklocalunicast: false ifaceLocalMulticast: false GlobalUni: false
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:0.0.0.0 -> 0.0.0.0:0
[2023-04-20 07:17:42]  DBG  Inode found in cache: 4.221µs 4893 66369 66369 192.168.0.151 22 192.168.0.205 58236
[2023-04-20 07:17:42]  DBG  [0] PID found 4893 [66369]
[2023-04-20 07:17:42]  DBG  UI is not running or busy, connected: false, running: false
[2023-04-20 07:17:42]  DBG  new connection tcp => 22:192.168.0.151 -> 192.168.0.205:58236 uid: 0
[2023-04-20 07:17:42]  DBG  [0/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58236 inode: 66369 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  [1/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:192.168.0.151 -> 192.168.0.205:58226 inode: 66303 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:192.168.0.151 -> 192.168.0.205:58226
[2023-04-20 07:17:42]  DBG  [2/3] outgoing connection uid: 0, 22:192.168.0.151 -> 192.168.0.205:58236 || netlink response: 22:0.0.0.0 -> 0.0.0.0:0 inode: 30940 - loopback: false multicast: false unspecified: true linklocalunicast: false ifaceLocalMulticast: false GlobalUni: false
[2023-04-20 07:17:42]  DBG  GetSocketInfo() invalid: 22:0.0.0.0 -> 0.0.0.0:0
[2023-04-20 07:17:42]  DBG  Inode found in cache: 4.589µs 4893 66369 66369 192.168.0.151 22 192.168.0.205 58236
[2023-04-20 07:17:42]  DBG  [0] PID found 4893 [66369]
[2023-04-20 07:17:42]  DBG  UI is not running or busy, connected: false, running: false
[2023-04-20 07:17:42]  DBG  new connection tcp => 39756:127.0.0.1 -> 127.0.0.1:6379 uid: 998
[2023-04-20 07:17:42]  DBG  [0/1] outgoing connection uid: 998, 39756:127.0.0.1 -> 127.0.0.1:6379 || netlink response: 39756:127.0.0.1 -> 127.0.0.1:6379 inode: 39716 - loopback: true multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: false
[2023-04-20 07:17:42]  DBG  new pid lookup took (2891): 3.330039ms
[2023-04-20 07:17:42]  DBG  [0] PID found 2891 [39716]
[2023-04-20 07:17:42]  DBG  UI is not running or busy, connected: false, running: false
[2023-04-20 07:17:43]  DBG  client.disconnect()
[2023-04-20 07:17:44]  DBG  client.disconnect()
[2023-04-20 07:17:44]  DBG  new connection tcp => 50384:192.168.0.151 -> 217.146.23.158:5938 uid: 0
[2023-04-20 07:17:44]  DBG  [0/1] outgoing connection uid: 0, 50384:192.168.0.151 -> 217.146.23.158:5938 || netlink response: 50384:192.168.0.151 -> 217.146.23.158:5938 inode: 29396 - loopback: false multicast: false unspecified: false linklocalunicast: false ifaceLocalMulticast: false GlobalUni: true
[2023-04-20 07:17:44]  DBG  new pid lookup took (2221): 4.903129ms
[2023-04-20 07:17:44]  DBG  [0] PID found 2221 [29396]
[2023-04-20 07:17:44]  DBG  UI is not running or busy, connected: false, running: false
[2023-04-20 07:17:45]  DBG  client.disconnect()
[2023-04-20 07:17:46]  DBG  client.disconnect()
[2023-04-20 07:17:47]  DBG  client.disconnect()
[2023-04-20 07:17:48]  DBG  client.disconnect()
^C
[2023-04-20 07:17:48]  IMP  Got signal: interrupt
[2023-04-20 07:17:48]  INF  Cleaning up ...
[2023-04-20 07:17:48]  DBG  worker channel closed 13
[2023-04-20 07:17:48]  DBG  stop monitoring firewall config file
[2023-04-20 07:17:48]  DBG  worker channel closed 3
[2023-04-20 07:17:48]  DBG  worker #3 exit
[2023-04-20 07:17:48]  INF  iptables, stop monitoring system fw rules
[2023-04-20 07:17:48]  DBG  worker channel closed 15
[2023-04-20 07:17:48]  DBG  worker #15 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 14
[2023-04-20 07:17:48]  DBG  worker #14 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 12
[2023-04-20 07:17:48]  DBG  worker #12 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 11
[2023-04-20 07:17:48]  DBG  worker #11 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 0
[2023-04-20 07:17:48]  DBG  worker #0 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 1
[2023-04-20 07:17:48]  DBG  worker #1 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 5
[2023-04-20 07:17:48]  DBG  worker #5 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 10
[2023-04-20 07:17:48]  DBG  worker #10 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 8
[2023-04-20 07:17:48]  DBG  worker #8 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 7
[2023-04-20 07:17:48]  DBG  worker #7 exit
[2023-04-20 07:17:48]  DBG  worker #13 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 9
[2023-04-20 07:17:48]  DBG  worker #9 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 4
[2023-04-20 07:17:48]  DBG  worker #4 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 6
[2023-04-20 07:17:48]  DBG  worker #6 exit
[2023-04-20 07:17:48]  DBG  worker channel closed 2
[2023-04-20 07:17:48]  DBG  worker #2 exit
[2023-04-20 07:17:48]  INF  exit checking iptables rules
```

</details>

___

I must add that since my original post, I have rebooted the machine, and the original error has changed:
`$ sudo /usr/bin/opensnitchd -rules-path /etc/opensnitchd/rules/ -debug`
```
[2023-04-20 07:20:35]  IMP  Starting opensnitch-daemon v1.5.2
[2023-04-20 07:20:35]  INF  Loading rules from /etc/opensnitchd/rules ...
[2023-04-20 07:20:35]  DBG  Starting 16 workers ...
[2023-04-20 07:20:35]  DBG  Worker #4 started.
ERROR: nfq_create_queue() queue not created
[2023-04-20 07:20:35]  DBG  Worker #9 started.
[2023-04-20 07:20:35]  DBG  Worker #6 started.
[2023-04-20 07:20:35]  DBG  Worker #11 started.
[2023-04-20 07:20:35]  DBG  Stats worker #2 started.
[2023-04-20 07:20:35]  DBG  Stats worker #3 started.
[2023-04-20 07:20:35]  DBG  Worker #8 started.
[2023-04-20 07:20:35]  WAR  Is opensnitchd already running?
[2023-04-20 07:20:35]  !!!  Error while creating queue #0: Error binding to queue: operation not permitted
```

I found it strange that I'd get an `operation not permitted` with `sudo` so I checked the service:
`$ systemctl status opensnitch.service`
```
● opensnitch.service - OpenSnitch is a GNU/Linux application firewall.
     Loaded: loaded (/lib/systemd/system/opensnitch.service; enabled; vendor preset: enabled)
     Active: active (running) since Thu 2023-04-20 10:10:50 EEST; 10min ago
       Docs: https://github.com/gustavo-iniguez-goya/opensnitch/wiki
    Process: 1228 ExecStartPre=/bin/mkdir -p /etc/opensnitchd/rules (code=exited, status=0/SUCCESS)
   Main PID: 1267 (opensnitchd)
      Tasks: 26 (limit: 38103)
     Memory: 70.0M
        CPU: 11.044s
     CGroup: /system.slice/opensnitch.service
             └─1267 /usr/bin/opensnitchd -rules-path /etc/opensnitchd/rules

Apr 20 10:10:50 stefan-mobile-desktop systemd[1]: Starting OpenSnitch is a GNU/Linux application firewall....
Apr 20 10:10:50 stefan-mobile-desktop systemd[1]: Started OpenSnitch is a GNU/Linux application firewall..
Apr 20 10:10:50 stefan-mobile-desktop opensnitchd[1267]: [2023-04-20 07:10:50]  IMP  Starting opensnitch-daemon v1.5.2
Apr 20 10:10:50 stefan-mobile-desktop opensnitchd[1267]: [2023-04-20 07:10:50]  INF  Loading rules from /etc/opensnitchd/rules ```

So I'm led to guess that all I needed to do after installing the daemon was just a reboot? (I usually put it to sleep between workdays)

**Comment by gustavo-iniguez-goya** [comment]
really strange. This is also how we managed to get it working here [#323](https://github.com/evilsocket/opensnitch/issues/323#issuecomment-753638436). The daemon doesn't require to reboot in order to work.

Here the libnetfilter_queue devs suggested just to ignore the errors of nfq_unbind_pf()
https://www.spinics.net/lists/netfilter/msg42063.html

I've installed 100ths of times on many distros and kernels and I've never run into this problem, so no idea. Maybe the next time would be interesting to have an strace output, to see the netlink communication messages with the kernel.

> Error binding to queue: operation not permitted

This error is usually caused when there's another process using the same queue, usually when opensnitchd is already running.
In that case it seems that systemd started the daemon, so it was already running when you launched it from the command line.
